extension/popup/vad.js

```diff
@@ -156,6 +156,9 @@ this.vad = (function() {
       log.debug("goCloud, why:", why);
       this.stopGum();
       // FIXME: maybe we need to signal the UI here?
+      if (why === "GoCloud finishedvoice") {
+        ui.setState("processing"); // REVIEW: not sure if this should send a message to popup.js or emit an event some other way?
```

Comment:
Technically the "right" way is to send the message to voice.js, and then to popup.js, and then to ui.js. I don't even know if this has a handle to voice.js (or the `Recorder` instance) at the moment. Maybe best to just have a TODO here.

Reply:
👍